### ISSUE TITLE
Issue setting temperature on 3 screens

### ISSUE THREAD

**Issue by iggl** [Tookmund/setcolortemperature#3]
Hi,

I've notice I can't set the temperature on my 3 screen display. I'm working on Debian "bullseye/sid", last updated and I've tested both the debian package and the code on this repo.

I run the command 
`$ ./sct 1000 `
to change color temperature on a really sensible way and the change is only effective on 2 displays. I can confirm the for loop seems to be running on the 3 screens, but looks like the 3rd screen do not effectively change the color.

If I perform this change on the code: 
```
iggl@LT02264-GNU:~/Downloads/setcolortemperature-master$ diff sct.c sct.c.orig 
70c70
< 	for (int c = 0; c <= res->ncrtc ; c++) {
---
> 	for (int c = 0; c < res->ncrtc; c++) {
72d71
< 		printf ("debug c = %d, res->ncrtc = %d, crtcxid = %d \n", c, res->ncrtc, crtcxid);

```
I get an error as the code seems to attempts to change the temperature on an non existing display, BUT funny enough the color temperature is now changed on the 3 monitors. Honestly I cant explain why :)

```
iggl@LT02264-GNU:~/Downloads/setcolortemperature-master$ ./sct 1000
debug c = 0, res->ncrtc = 3, crtcxid = 63 
debug c = 1, res->ncrtc = 3, crtcxid = 64 
debug c = 2, res->ncrtc = 3, crtcxid = 65 
debug c = 3, res->ncrtc = 3, crtcxid = 66 
X Error of failed request:  BadRRCrtc (invalid Crtc parameter)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  22 (RRGetCrtcGammaSize)
  Crtc id in failed request: 0x42
  Serial number of failed request:  18
  Current serial number in output stream:  18
```

**Comment by Tookmund** [comment]
Thanks for reporting this, it's actually been on my backlog of things to fix for...oh wow it's been a long time.
This is probably the same issue reported [in Ubuntu](https://bugs.launchpad.net/ubuntu/+source/setcolortemperature/+bug/1888632).
As mentioned there, a better maintained fork exists that should fix your issue:
https://github.com/faf0/sct

I keep meaning to incorporate their changes upstream, but just haven't got around to it yet.
Likely they should be the upstream of the debian package at this point, since it's much better maintained.

**Comment by iggl** [comment]
Hi.

I understand lack of time is hard. If you think is a good idea to change the fork, maybe you can contact the Debian maintainers to see what they think about.

The patch was quite simple. After this the sct command works as intended. I guess without close the dpy display, the last changes are not committed and updated before the program finishes.

iggl@LT02264-GNU:~/Downloads/setcolortemperature-master$ diff sct.c sct.c.orig
86,89d85
< 
< 	XFree(res);
< 	XCloseDisplay(dpy);
<

**Comment by Tookmund** [comment]
Thanks for the patch! I've applied it since it seemed straightforward, though I don't have three displays to test with.
I'm actually the current maintainer of the sct debian package, but just discovered that someone's already looking into [packaging xsct](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=984779).

I reached out to him and apparently the best process is to [orphan setcolortemperature](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=984936), and then have xsct eventually ship a transitional package to migrate old users to it, so I did that.
Not in time for bullseye unfortunately, but it should be in the next release of debian.
